frontend/src/features/map_planning/layers/shade/ShadeLayer.tsx

```diff
@@ -0,0 +1,239 @@
+
+      function placeNewShading(point: Vector2d) {
+        executeAction(
+          new CreateShadingAction({
+            id: uuid.v4(),
+            layerId: shadeLayerId,
+            addDate: timelineDate,
+            shade: selectedShadeForNewShading ?? Shade.NoShade,
+            geometry: ringGeometryAroundPoint({ ...point, srid: DEFAULT_SRID }, 8, 200),
+          }),
+        );
+      }
+    },
+    [selectedShadeForNewShading, executeAction],
+  );
+
+  const handleSelectShading: KonvaEventListener<Konva.Stage, MouseEvent> = useCallback(() => {
+    const currentLayer = useMapStore.getState().untrackedState.selectedLayer;
+    const shadingManipulationState =
+      useMapStore.getState().untrackedState.layers.shade.selectedShadingEditMode;
+
+    if (
+      typeOfLayer(currentLayer) !== LayerType.Shade ||
+      shadingManipulationState !== 'inactive' ||
+      selectedShadeForNewShading !== null
+    ) {
+      return;
+    }
+
+    const selectedShadings = (foundPlantings: ShadingDto[], konvaNode: Node) => {
+      const shadingNode = isShading(konvaNode);
+      return shadingNode ? [...foundPlantings, shadingNode] : [foundPlantings];
+    };
+
+    const transformer = useMapStore.getState().transformer.current;
+    const shadings = transformer?.nodes().reduce(selectedShadings, []);
+
+    if (shadings?.length) {
+      useMapStore.getState().shadeLayerSelectShadings(shadings);
+    }
+  }, [selectedShadeForNewShading]);
+
+  const handleUnselectShading = useCallback(
+    (e: Konva.KonvaEventObject<MouseEvent>) => {
+      const currentLayer = useMapStore.getState().untrackedState.selectedLayer;
+      const shadingManipulationState =
+        useMapStore.getState().untrackedState.layers.shade.selectedShadingEditMode;
+
+      if (
+        typeOfLayer(currentLayer) !== LayerType.Shade ||
+        shadingManipulationState !== 'inactive' ||
+        selectedShadeForNewShading !== null
+      ) {
+        return;
+      }
+
+      if (!isShading(e.target)) {
+        shadeLayerSelectShading(null);
+      }
+    },
+    [selectedShadeForNewShading, shadeLayerSelectShading],
+  );
+
+  useEffect(() => {
+    if (!props.listening) {
+      return;
+    }
+
+    useMapStore.getState().stageRef.current?.on('click.placeShading', handlePlaceShading);
+    useMapStore.getState().stageRef.current?.on('click.unselectShading', handleUnselectShading);
+    useMapStore.getState().stageRef.current?.on('mouseup.selectShading', handleSelectShading);
+
+    return () => {
+      useMapStore.getState().stageRef.current?.off('click.placeShading');
+      useMapStore.getState().stageRef.current?.off('click.unselectShading');
+      useMapStore.getState().stageRef.current?.off('mouseup.selectShading');
+    };
+  }, [handlePlaceShading, handleSelectShading, handleUnselectShading, props.listening]);
+
+  /*
+    This is an ugly workaround for a pretty annoying bug involving the Konva transformer.
```

Comment:
did you create an issue for this? maybe it could be intersetng to investigate this workaround later.

Reply:
Not yet, but I will create one.

Reply:
still open?

Reply:
@andreicristian97 can you help out here? These is an issue only effecting the dev branch.

Reply:
@badnames still open

Reply:
Created new issue to cover this remaining point: https://github.com/ElektraInitiative/PermaplanT/issues/1301